unicon_runner/executor/variants/sandbox/executor.py

```diff
@@ -42,33 +46,73 @@ async def _execute(self, request: Request, request_id: str, folder_path: str) ->
             f.write("")
 
         with open(os.path.join(folder_path, "requirements.txt"), "w") as f:
-            if "requirements" in request.environment.extra_options:
+            if (
+                request.environment.extra_options
+                and "requirements" in request.environment.extra_options
+            ):
                 f.write(request.environment.extra_options["requirements"])
 
+        mem_limit_mb: int = request.environment.memory_limit * 1024
+        time_limit_secs: int = request.environment.time_limit
+
+        python_version: str = "3.11.9"
+        if request.environment.extra_options:
+            python_version = request.environment.extra_options.get("version", python_version)
+
         # 2. Cd into temp folder and run uv sync && uv run entry
-        proc = await asyncio.create_subprocess_shell(
-            f"UV_CONCURRENT_INSTALLS=1 {self.INSTALL_SCRIPT} {folder_path}",
+        install_proc = await asyncio.create_subprocess_shell(
+            shlex.join([self.INSTALL_SCRIPT, folder_path, python_version]),
             stdout=asyncio.subprocess.PIPE,
             stderr=asyncio.subprocess.PIPE,
+            # NOTE: We need to unset VIRTUAL_ENV to prevent uv from using it
+            env={**os.environ, "VIRTUAL_ENV": ""},
```

Comment:
will add error handling for install subprocess in a future PR. there is a chance errors or warnings are raised here but not caught which may have downstream effects on execution subprocess